File: crud/src/main/java/fr/cicco/crud/entity/Menu.java
package fr.cicco.crud.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;    
    
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Builder
public class Menu   {
    @Id()
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Double prix;
    private String nom;
    private String photo;
    private String description;
    @ManyToMany
    @JoinTable(name="accompagnement_menu",
            joinColumns = @JoinColumn(name="menu_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name="accompagnement_id", referencedColumnName = "id")
    )
    private List<Accompagnement> accompagnementList;
    @ManyToMany
    @JoinTable(name="menu_pizza",
            joinColumns = @JoinColumn(name="menu_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name="pizza_id", referencedColumnName = "id")
    )
    private List<Pizza> pizzaList;
    @OneToMany(mappedBy = "menu")
    private List<CommandeMenu> commandeMenuList;

}

File: crud/springCLI/python_entities_template/menu.py

from ..datas.autocompletion import *

name = 'Menu'

entity = {
    
    'Menu' : {
            
        'id':{
            Type: Long,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'prix':{
            Type: Double,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'nom':{
            Type: String,
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'photo':{
            Type: "String",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'description':{
            Type: "String",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'accompagnementList':{
            Type: "List<Accompagnement>",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'pizzaList':{
            Type: "List<Pizza>",
            options:{
                'dto':[
                    {'exists':True, 'fields':[]}
                ]
        
            }
        },

        'commandeMenuList':{
            Type: "List<CommandeMenu>",
            options:{
                'dto':[
                ]
        
            }
        },

        }

}


